language/move-prover/bytecode/src/spec_instrumentation.rs

```diff
@@ -515,6 +515,24 @@ impl<'a> Instrumenter<'a> {
                 );
                 self.builder.emit_with(move |id| Prop(id, Assume, exp));
             }
+
+            // Emit the events in the `emits` specs of the callee.
+            for (_, msg, handle, cond) in std::mem::take(&mut callee_spec.emits) {
+                let temp_msg = self.builder.emit_let(msg).0;
+                let temp_handle = self.builder.emit_let(handle).0;
```

Comment:
@wrwg, after rebasing on your `$Mutation` PR, I get the following error:
```
prover_emits.bpl(1883,47): Error: invalid type for argument 0 in application of $SelectField: $Mutation (expected: $Value)
```
in the line of 
```
$es := $ExtendEventStore($es, $SelectField($t5, $Event_EventHandle_guid), $t4);
```

I think `$t5` needs to be wrapped with `$Dereference(...)` in the generated Boogie code. Which part of my PR should be corrected?

Reply:
You need to change the Boogie procedures. The API is now to take and return a $Mutation where there was a value before. Checkout other Boogie procedures like those for vectors. The translation is correct but different.